src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultCredentialRetriever.java

```diff
@@ -57,4 +62,55 @@ public static KeyVaultCredentials getCredentialById(String credentialID, Run<?,
 
         return credential;
     }
+
+    public static KeyVaultCredentials getCredentialById(String credentialID) {
```

Comment:
from what I can see this should be able to be refactored to be an overload and null handled in the other method.

seems to be a lot of duplication here

Reply:
You mean the `KeyVaultCredentials getCredentialById(String credentialID, Run<?, ?> build)` method?

Reply:
yes

Reply:
The method `CredentialsProvider.findCredentialById(credentialID, IdCredentials.class, build);` used to retrieve credentials  in that one will try to fetch all the credentials including this provider. This will finally cause a dead loop. Is it make sense to only fetch system credentials for that method too?

Reply:
There should be some specific difference for credential scope or permission issues for these two methods.

Reply:
ah I see